### ISSUE TITLE
Charmcraft analyze should try to identify a License

### ISSUE THREAD

**Issue by heitorPB** [canonical/charmcraft#464]
`charmcraft analyze` should try to identify the license used for the charm.

This could be done by looking for a file `LICENSE` and matching its content to the most famous ones. This information could then be propagated to `juju info <charm>` and Charmhub.

**Comment by sergiusens** [comment]
It might be a better idea to use metadata.yaml and introduce SPDX, would you mind creating a forum post on discourse for this?

**Comment by sergiusens** [comment]
It would be very hard to parse a license file without building a checksum database to be certain of our match. There can be legal implications due to this. A wider group discussion on discourse would be appropriate.

**Comment by heitorPB** [comment]
Discourse post: https://discourse.charmhub.io/t/juju-should-know-about-charm-licenses/4984
